### DISCUSSION TITLE
modifying the surfaces

### DISCUSSION THREAD

**Discussion by sevamoo** [cityjson/specs#113] [Q&A] [answered]
Hi
Thanks a lot for the great work! So easy to work with Cityjson 3d building data! 
I apologize since what I am writing here is not an issue, but rather questions related to features/functionalities to modify the surfaces.
While I found it extremely easy to perform analytics on different aspects of geometries, I am wondering if there are some functionalities to modify the geometries in cjio?
  
The use case is how to modify the geometry of walls and ground surfaces by intersecting the initial building with the DEM surface. And of course ideally to do this programmatically at scale. 
Here you can see two snapshots of a building

![Screenshot 2022-03-21 at 16 54 49](https://user-images.githubusercontent.com/516107/159301978-464cfd07-c7ca-4cdc-afed-45a82c809f38.png)

![Screenshot 2022-03-21 at 16 58 02](https://user-images.githubusercontent.com/516107/159301956-ff693b7a-d5c2-44f1-8be0-f43edca8949d.png)

I would appreciate any advice from anyone and if I manage to find a solution will post it here.

Best
Vahid

**Comment by liberostelios** [comment, marked as answer]
Hi Vahid,

You are right, we don't currently have discussions here so we can only discuss things as issue. That's absolutely fine, though.

You may want to check [pyvista](https://docs.pyvista.org/), a Python library that allows you to do a lot of things regarding 3D geometric manipulation and it's awesome enough to persist semantics! You are probably looking to something like [this](https://docs.pyvista.org/examples/01-filter/clipping-with-surface.html).

Parsing CityJSON in pyvista is relatively straightforward, although you might have to  triangulate in the process. We have implemented a parser as part of our [3D Building Metrics](https://github.com/tudelft3d/3d-building-metrics) toolkit and you may want to "borrow" our implementation [here](/blob/cf80cc241f7fceee2c25ac99efe6e48fa0ae8095/cityjson.py#L71) which loads semantic surface types (don't worry, as with everything, it's copyleft). You can check the notebooks for some experimental/demo code on what we do with CityJSON and pyvista.

**Comment by sevamoo** [comment]
Hi Stelios 
Thanks a lot for this great answer. I had seen Pyvista in a presentation by you but gave up since I couldn't install some dependencies. But need to do it now anyway! 
I will go through the links in detail, but the last [link](https://github.com/blob/cf80cc241f7fceee2c25ac99efe6e48fa0ae8095/cityjson.py#L71) was apparently broken.

Regarding the task at hand: This is a part of my personal project, with the potential to be a startup project.
I have really high respect for the work you do in the 3D Geoinformation Group at TUD. 
If you are interested I would be more than happy to tell you more about the details of my project.
Best
Vahid
